### PR TITLE
fix: Fix duplicate column output and panic for `include_file_paths`

### PR DESCRIPTION
**Pull request by nameexhaustion**
Fixes https://github.com/pola-rs/polars/issues/18219#issuecomment-2295184914

Also fixes a panic for batched CSV and Parquet (fixes https://github.com/pola-rs/polars/issues/18257)

```rust
py-polars/tests/unit/io/test_scan.py ..thread '<unnamed>' panicked at crates/polars-core/src/chunked_array/mod.rs:643:9:
index: 10 out of bounds for len: 1
```

### CODE REVIEW

File: crates/polars-core/src/chunked_array/ops/mod.rs [hunk 1]
```diff
@@ -461,7 +461,7 @@ pub trait ChunkFilter<T: PolarsDataType> {
 /// Create a new ChunkedArray filled with values at that index.
 pub trait ChunkExpandAtIndex<T: PolarsDataType> {
     /// Create a new ChunkedArray filled with values at that index.
-    fn new_from_index(&self, length: usize, index: usize) -> ChunkedArray<T>;
+    fn new_from_index(&self, index: usize, length: usize) -> ChunkedArray<T>;
```
Comment (nameexhaustion) [inline crates/polars-core/src/chunked_array/ops/mod.rs:464]: index and length were flipped on the trait definition

File: crates/polars-core/src/frame/mod.rs [hunk 2]
```diff
@@ -1171,8 +1171,15 @@ impl DataFrame {
     /// # Safety
     /// The caller must ensure `column.len() == self.height()` .
     pub unsafe fn with_column_unchecked(&mut self, column: Series) -> &mut Self {
-        self.get_columns_mut().push(column);
-        self
+        #[cfg(debug_assertions)]
+        {
+            return self.with_column(column).unwrap();
```
Comment (nameexhaustion) [inline crates/polars-core/src/frame/mod.rs:1176]: drive-by - enable checks here in debug mode to prevent similar mistakes

File: crates/polars-mem-engine/src/executors/scan/parquet.rs [hunk 3]
```diff
@@ -210,20 +209,6 @@ impl ParquetExec {
                                 )?
                                 .finish()?;
 
-                            if let Some(col) = &self.file_options.include_file_paths {
```
Comment (nameexhaustion) [inline crates/polars-mem-engine/src/executors/scan/parquet.rs:213]: This was duplicate from a very old implementation approach that didn't push `include_file_paths` into the actual reader.

**Review by ritchie46 — APPROVED** [review]
